app/controllers/Application.scala

```diff
@@ -138,7 +138,7 @@ class Application @Inject() (
 
   def onHandlerNotFound(route: String) = Action { implicit request =>
     if (route.endsWith("/")) {
-      MovedPermanently(request.path.take(request.path.length - 1))
+      MovedPermanently("//" + request.host + "/" + request.path.take(request.path.length - 1).dropWhile(_ == '/'))
```

Comment:
You're right, I removed `request.host` from the redirect URL.
Regarding the regex, feel free to fine-tune the PR to your needs.